





















































<img src="http://imgur.com/1ZcRyrc.png" style="float: left; margin: 20px; height: 55px">

# Project 2: Analyzing Chipotle Data

_Author: Joseph Nelson (DC)_

---

For Project 2, you will complete a series of exercises exploring [order data from Chipotle](https://github.com/TheUpshot/chipotle), compliments of _The New York Times'_ "The Upshot."

For these exercises, you will conduct basic exploratory data analysis (Pandas not required) to understand the essentials of Chipotle's order data: how many orders are being made, the average price per order, how many different ingredients are used, etc. These allow you to practice business analysis skills while also becoming comfortable with Python.

---

## Basic Level

### Part 1: Read in the file with `csv.reader()` and store it in an object called `file_nested_list`.

Hint: This is a TSV (tab-separated value) file, and `csv.reader()` needs to be told [how to handle it](https://docs.python.org/2/library/csv.html).

In [1]:
import csv
from collections import namedtuple   # Convenient to store the data rows

#DATA_FILE = './data/chipotle.tsv' - untagged because reading in data from harddrive

In [2]:
# calling up tsv file and storing it under 'file_nested_list':
with open('C:/users/rickk/Documents/DataScience/GeneralAssembly/Project02/data/chipotle.tsv','r') as file_nested_list:
    lines = file_nested_list.readlines()

In [3]:
# removing the '\n' at the end of each line
cleaned_lines = []
for l in lines:
    cleaned_lines.append(l.replace('\n',''))
cleaned_lines

# removing the '$' for the pricing information, so that it is easier to calculate with later on
cleaned_lines_2 = []
for l in cleaned_lines:
    cleaned_lines_2.append(l.replace('$',''))
cleaned_lines_2

['order_id\tquantity\titem_name\tchoice_description\titem_price',
 '1\t1\tChips and Fresh Tomato Salsa\tNULL\t2.39 ',
 '1\t1\tIzze\t[Clementine]\t3.39 ',
 '1\t1\tNantucket Nectar\t[Apple]\t3.39 ',
 '1\t1\tChips and Tomatillo-Green Chili Salsa\tNULL\t2.39 ',
 '2\t2\tChicken Bowl\t[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]\t16.98 ',
 '3\t1\tChicken Bowl\t[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]\t10.98 ',
 '3\t1\tSide of Chips\tNULL\t1.69 ',
 '4\t1\tSteak Burrito\t[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]\t11.75 ',
 '4\t1\tSteak Soft Tacos\t[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]\t9.25 ',
 '5\t1\tSteak Burrito\t[Fresh Tomato Salsa, [Rice, Black Beans, Pinto Beans, Cheese, Sour Cream, Lettuce]]\t9.25 ',
 '5\t1\tChips and Guacamole\tNULL\t4.45 ',
 '6\t1\tChicken Crispy Tacos\t[Roasted Chili Corn Salsa, [Fajita Vegetable

In [ ]:
# if I wanted to store the lines in namedtupple... how would I best do that?
    # example: 
    # p = Point(X=11, y=22) --- p._fields >> ('x','y')
    # Color = namedtuple('Color', 'red green blue')
    # Pixel = namedtuple('Pixel', Point._fields + Color._fields)
    # Pixel(11,22,128,255,0)
    # >> yields: Pixel(x=11, y=22, red=128, green=255, blue=0)
# useful if you want to call a specific element of namedtuple: getattr(p,'x') returns 11
# << this last point is similar to specifying the row position, but instead of having to know row[...] you call the 'x'

### Part 2: Separate `file_nested_list` into the `header` and the `data`.


In [4]:
# separate the header, which is line [0]
header = cleaned_lines_2[0]

In [5]:
print (header)

order_id	quantity	item_name	choice_description	item_price


In [6]:
# split the header into the individual categories
split_header = header.split('\t')

In [7]:
split_header

['order_id', 'quantity', 'item_name', 'choice_description', 'item_price']

In [8]:
# separate out the data, which is in lines [1:] "lines one and following"
data = cleaned_lines_2[1:]

In [9]:
# split the data into individual columns, which is needed for further analysis
split_data = []
for d in data:
    split_data.append(d.split('\t'))

In [10]:
split_data

[['1', '1', 'Chips and Fresh Tomato Salsa', 'NULL', '2.39 '],
 ['1', '1', 'Izze', '[Clementine]', '3.39 '],
 ['1', '1', 'Nantucket Nectar', '[Apple]', '3.39 '],
 ['1', '1', 'Chips and Tomatillo-Green Chili Salsa', 'NULL', '2.39 '],
 ['2',
  '2',
  'Chicken Bowl',
  '[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]',
  '16.98 '],
 ['3',
  '1',
  'Chicken Bowl',
  '[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]',
  '10.98 '],
 ['3', '1', 'Side of Chips', 'NULL', '1.69 '],
 ['4',
  '1',
  'Steak Burrito',
  '[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]',
  '11.75 '],
 ['4',
  '1',
  'Steak Soft Tacos',
  '[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]',
  '9.25 '],
 ['5',
  '1',
  'Steak Burrito',
  '[Fresh Tomato Salsa, [Rice, Black Beans, Pinto Beans, Cheese, Sour Cream, Lettuce]]',
  '9.25 '],
 ['5', '1', 'Chips and Guacamole', 'NULL', 

---

## Intermediate Level

### Part 3: Calculate the average price of an order.

Hint: Examine the data to see if the `quantity` column is relevant to this calculation.

Hint: Think carefully about the simplest way to do this!

In [11]:
# checking on quantity in second column col[1] - if all quantities are equal to 1, it does not matter
quantity = []
for row in split_data:
    quantity.append(float(row[1])) # turning strings into floats so they can be used for arithmetic operations
print(quantity)
len(quantity)

[1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 2.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,

4622

Data shows that quantities are anywhere between 1 and 10. Quantities do matter for the calculation of average prices per order.

In [12]:
# extract the amount paid per order, i.e. per line in the dataset
paid = []
for row in split_data:
    paid.append(float(row[4])) # turning strings into floats so they can be used for arithmetic operations
print(paid)
len(paid)

[2.39, 3.39, 3.39, 2.39, 16.98, 10.98, 1.69, 11.75, 9.25, 9.25, 4.45, 8.75, 8.75, 11.25, 4.45, 2.39, 8.49, 8.49, 2.18, 8.75, 4.45, 8.99, 3.39, 10.98, 3.39, 2.39, 8.49, 8.99, 1.09, 8.49, 2.39, 8.99, 1.69, 8.99, 1.09, 8.75, 8.75, 4.45, 2.95, 11.75, 2.15, 4.45, 11.25, 11.75, 8.75, 10.98, 8.99, 3.39, 8.99, 3.99, 8.99, 2.18, 10.98, 1.09, 8.99, 2.39, 9.25, 11.25, 11.75, 2.15, 4.45, 9.25, 11.25, 8.75, 8.99, 8.99, 3.39, 8.99, 10.98, 8.99, 1.69, 8.99, 3.99, 8.75, 4.45, 8.75, 8.75, 2.15, 8.75, 11.25, 2.15, 9.25, 8.75, 8.75, 9.25, 8.49, 8.99, 1.09, 9.25, 2.95, 11.75, 11.75, 9.25, 11.75, 4.45, 9.25, 4.45, 11.75, 8.75, 8.75, 4.45, 8.99, 8.99, 3.99, 8.49, 3.39, 8.99, 1.09, 9.25, 4.45, 8.75, 2.95, 4.45, 2.39, 8.49, 8.99, 8.49, 1.09, 8.99, 3.99, 8.75, 9.25, 4.45, 11.25, 4.45, 8.99, 1.09, 9.25, 2.95, 4.45, 11.75, 4.45, 8.49, 2.39, 10.98, 22.5, 11.75, 4.45, 11.25, 4.45, 11.25, 4.45, 11.25, 11.25, 11.75, 9.25, 4.45, 11.48, 17.98, 1.69, 17.5, 4.45, 8.49, 2.39, 17.5, 4.45, 4.45, 11.25, 11.75, 10.98, 8.49, 

4622

In [13]:
# the two prior lists have the same length
# we can divide each $-price in list 'paid' by the corresponding item in list 'quantity'
average_price = []
for i in range(0,len(paid)):
    price = paid[i]/quantity[i]
    average_price.append(price)
print(average_price)

[2.39, 3.39, 3.39, 2.39, 8.49, 10.98, 1.69, 11.75, 9.25, 9.25, 4.45, 8.75, 8.75, 11.25, 4.45, 2.39, 8.49, 8.49, 1.09, 8.75, 4.45, 8.99, 3.39, 10.98, 3.39, 2.39, 8.49, 8.99, 1.09, 8.49, 2.39, 8.99, 1.69, 8.99, 1.09, 8.75, 8.75, 4.45, 2.95, 11.75, 2.15, 4.45, 11.25, 11.75, 8.75, 10.98, 8.99, 3.39, 8.99, 3.99, 8.99, 1.09, 10.98, 1.09, 8.99, 2.39, 9.25, 11.25, 11.75, 2.15, 4.45, 9.25, 11.25, 8.75, 8.99, 8.99, 3.39, 8.99, 10.98, 8.99, 1.69, 8.99, 3.99, 8.75, 4.45, 8.75, 8.75, 2.15, 8.75, 11.25, 2.15, 9.25, 8.75, 8.75, 9.25, 8.49, 8.99, 1.09, 9.25, 2.95, 11.75, 11.75, 9.25, 11.75, 4.45, 9.25, 4.45, 11.75, 8.75, 8.75, 4.45, 8.99, 8.99, 3.99, 8.49, 3.39, 8.99, 1.09, 9.25, 4.45, 8.75, 2.95, 4.45, 2.39, 8.49, 8.99, 8.49, 1.09, 8.99, 3.99, 8.75, 9.25, 4.45, 11.25, 4.45, 8.99, 1.09, 9.25, 2.95, 4.45, 11.75, 4.45, 8.49, 2.39, 10.98, 11.25, 11.75, 4.45, 11.25, 4.45, 11.25, 4.45, 11.25, 11.25, 11.75, 9.25, 4.45, 11.48, 8.99, 1.69, 8.75, 4.45, 8.49, 2.39, 8.75, 4.45, 4.45, 11.25, 11.75, 10.98, 8.49, 1

In [14]:
# overall average
total_average = sum(average_price) / len(average_price)
total_average

7.084424491562234

The overall average price per order is about **US$ 7.08** (taking the quantities ordered into account)

### Part 4: Create a list (or set) named `unique_sodas` containing all of unique sodas and soft drinks that Chipotle sells.

Note: Just look for `'Canned Soda'` and `'Canned Soft Drink'`, and ignore other drinks like `'Izze'`.

In [15]:
# create a list and check in the respective position in each row, i.e. third column row[2] for the two key words
# if the drink is already part of the list, it does not have to be passed into the list again and can be skipped
unique_sodas = []
for row in split_data:
    if row[2] == 'Canned Soda' or row[2] == 'Canned Soft Drink':
        if row[3] in unique_sodas: # skipping drink if already in list
            None
        else:
            unique_sodas.append(row[3])
    else:
        None
print(unique_sodas)

['[Sprite]', '[Dr. Pepper]', '[Mountain Dew]', '[Diet Dr. Pepper]', '[Coca Cola]', '[Diet Coke]', '[Coke]', '[Lemonade]', '[Nestea]']


In [16]:
# for kicks created a new dictionary that uses the brand names as keys and counts the occurence as values
# ignoring the quantity of orders here; only counting frequency
soda_count = {}
for brand in unique_sodas:
    soda_count[brand] = []

for i in range(0,len(unique_sodas)):
    count = 0
    for row in split_data:
        if row[3] == unique_sodas[i]:
            count += 1
    soda_count[unique_sodas[i]].append(count)
print (soda_count)

{'[Sprite]': [77], '[Dr. Pepper]': [18], '[Mountain Dew]': [15], '[Diet Dr. Pepper]': [13], '[Coca Cola]': [26], '[Diet Coke]': [134], '[Coke]': [123], '[Lemonade]': [33], '[Nestea]': [20]}


In [17]:
# sorting the soda count from lowest value to highest value
import operator
sorted_soda = sorted(soda_count.items(), key=operator.itemgetter(1)) # itemgetter(1) sorts on values; itemgetter(0) on keys
print(sorted_soda)

[('[Diet Dr. Pepper]', [13]), ('[Mountain Dew]', [15]), ('[Dr. Pepper]', [18]), ('[Nestea]', [20]), ('[Coca Cola]', [26]), ('[Lemonade]', [33]), ('[Sprite]', [77]), ('[Coke]', [123]), ('[Diet Coke]', [134])]


Apparently **Coke** and **Diet Coke** are the most ordered canned beverages at Chipotle with **123** and **134** orders respectively.

In [18]:
# let's do the same, but take the order quantity into consideration
soda_count_all = {}
for brand in unique_sodas:
    soda_count_all[brand] = []

for i in range(0,len(unique_sodas)):
    count = 0
    for row in split_data:
        if row[3] == unique_sodas[i]:
            count += 1*float(row[1])
    soda_count_all[unique_sodas[i]].append(count)
print (soda_count_all)

{'[Sprite]': [89.0], '[Dr. Pepper]': [21.0], '[Mountain Dew]': [18.0], '[Diet Dr. Pepper]': [14.0], '[Coca Cola]': [32.0], '[Diet Coke]': [159.0], '[Coke]': [143.0], '[Lemonade]': [36.0], '[Nestea]': [20.0]}


In [19]:
# sorting soda count all from lowest value to highest value
sorted_soda_all = sorted(soda_count_all.items(), key=operator.itemgetter(1))
print (sorted_soda_all)

[('[Diet Dr. Pepper]', [14.0]), ('[Mountain Dew]', [18.0]), ('[Nestea]', [20.0]), ('[Dr. Pepper]', [21.0]), ('[Coca Cola]', [32.0]), ('[Lemonade]', [36.0]), ('[Sprite]', [89.0]), ('[Coke]', [143.0]), ('[Diet Coke]', [159.0])]


Biggest difference for Sprite, Coke and Diet Coke - apparently customers like to order multiple of these together (as a group). The volumes increase for:
    \tSprite: 77 to 89
    \tCoke: 123 to 143
    \tDiet Coke: 134 to 159

---

## Advanced Level


### Part 5: Calculate the average number of toppings per burrito.

Note: Let's ignore the `quantity` column to simplify this task.

Hint: Think carefully about the easiest way to count the number of toppings!


In [20]:
toppings = []

for row in split_data:
    if "Burrito" in row[2]:
        toppings.append(row[3])
toppings

['[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]',
 '[Fresh Tomato Salsa, [Rice, Black Beans, Pinto Beans, Cheese, Sour Cream, Lettuce]]',
 '[Tomatillo-Green Chili Salsa (Medium), [Pinto Beans, Cheese, Sour Cream]]',
 '[Fresh Tomato Salsa (Mild), [Black Beans, Rice, Cheese, Sour Cream, Lettuce]]',
 '[[Fresh Tomato Salsa (Mild), Tomatillo-Green Chili Salsa (Medium), Tomatillo-Red Chili Salsa (Hot)], [Rice, Cheese, Sour Cream, Lettuce]]',
 '[[Tomatillo-Green Chili Salsa (Medium), Tomatillo-Red Chili Salsa (Hot)], [Pinto Beans, Rice, Cheese, Sour Cream, Guacamole, Lettuce]]',
 '[[Tomatillo-Green Chili Salsa (Medium), Roasted Chili Corn Salsa (Medium)], [Black Beans, Rice, Sour Cream, Lettuce]]',
 '[Tomatillo-Green Chili Salsa (Medium), [Pinto Beans, Rice, Cheese, Sour Cream]]',
 '[[Roasted Chili Corn Salsa (Medium), Fresh Tomato Salsa (Mild)], [Rice, Black Beans, Sour Cream]]',
 '[Fresh Tomato Salsa, [Rice, Pinto Beans, C

In [21]:
# from our list of toppings, the easiest way is to calculate the ',' as some toppings consist of multiple words
# in every order_description, the toppings are separated by a ',' irrespective of the number of words used to describe topping

toppings_count = []

for row in toppings:
    count = 1 # need to start the counter at 1, because there is +1 item vs the sum of commas in each row
    for item in row:
        if item == ',':
            count += 1
        else:
            pass
    toppings_count.append(count)
print(toppings_count)

[8, 7, 4, 6, 7, 8, 6, 5, 5, 7, 7, 4, 6, 6, 8, 8, 4, 7, 3, 7, 4, 6, 5, 6, 6, 5, 5, 3, 4, 5, 6, 5, 6, 5, 6, 6, 6, 5, 6, 5, 5, 5, 5, 5, 6, 5, 5, 9, 8, 6, 4, 6, 6, 2, 5, 5, 5, 4, 6, 4, 5, 7, 6, 6, 5, 4, 4, 4, 6, 7, 4, 7, 5, 6, 5, 7, 5, 6, 6, 3, 5, 8, 6, 4, 5, 4, 4, 4, 5, 7, 3, 4, 5, 8, 2, 7, 6, 5, 7, 6, 4, 5, 6, 4, 4, 5, 7, 7, 5, 4, 5, 6, 4, 6, 7, 5, 3, 7, 2, 6, 6, 4, 6, 6, 6, 6, 5, 6, 5, 8, 6, 8, 4, 6, 5, 7, 5, 7, 5, 7, 7, 5, 5, 7, 7, 8, 6, 7, 2, 4, 3, 5, 6, 2, 7, 6, 5, 7, 4, 5, 4, 5, 5, 2, 6, 5, 1, 1, 5, 1, 6, 7, 5, 5, 7, 6, 5, 5, 6, 4, 7, 7, 6, 6, 5, 4, 4, 7, 1, 4, 6, 5, 3, 5, 4, 5, 4, 5, 5, 5, 7, 6, 5, 8, 4, 3, 4, 6, 7, 7, 5, 6, 5, 8, 5, 7, 5, 6, 1, 5, 6, 3, 5, 7, 5, 3, 1, 6, 5, 6, 5, 6, 4, 7, 4, 4, 5, 7, 6, 4, 7, 7, 6, 7, 5, 5, 5, 5, 5, 5, 6, 5, 5, 5, 4, 4, 5, 5, 5, 5, 6, 6, 5, 6, 4, 4, 5, 5, 1, 6, 6, 5, 6, 7, 6, 6, 4, 6, 7, 5, 6, 6, 6, 5, 5, 4, 6, 4, 5, 5, 5, 8, 1, 5, 7, 5, 5, 5, 3, 7, 1, 4, 6, 5, 5, 5, 5, 6, 2, 3, 5, 5, 7, 6, 7, 5, 7, 6, 6, 6, 7, 4, 6, 6, 4, 6, 7, 5, 7, 7, 7, 5, 6, 

In [22]:
toppings_total_average = sum(toppings_count) / len (toppings_count)
toppings_total_average

5.395051194539249

On average, a burrito comes with **5.4 toppings** per order (ignoring quantity of burritos ordered, because that would not change the outcome dramatically)

### Part 6: Create a dictionary. Let the keys represent chip orders and the values represent the total number of orders.

Expected output: `{'Chips and Roasted Chili-Corn Salsa': 18, ... }`

Note: Please take the `quantity` column into account!

Optional: Learn how to use `.defaultdict()` to simplify your code.

In [23]:
# replicating the *fun project* from earlier that was counting sodas by brand with chips

unique_chips = []
for row in split_data:
    if "Chips" in row[2]:
        if row[2] in unique_chips:
            None
        else:
            unique_chips.append(row[2])
print (unique_chips)
print (len(unique_chips))

chips_count_all = {}
for combos in unique_chips: #let's refer to different chips + other as combos
    chips_count_all[combos] = []

for i in range(0,len(unique_chips)):
    count = 0
    for row in split_data:
        if row[2] == unique_chips[i]:
            count += 1*float(row[1]) # multiplying count with quantity in that row to get the total number of orders
    chips_count_all[unique_chips[i]].append(count)

print (chips_count_all)

['Chips and Fresh Tomato Salsa', 'Chips and Tomatillo-Green Chili Salsa', 'Side of Chips', 'Chips and Guacamole', 'Chips and Tomatillo Green Chili Salsa', 'Chips', 'Chips and Tomatillo Red Chili Salsa', 'Chips and Roasted Chili-Corn Salsa', 'Chips and Roasted Chili Corn Salsa', 'Chips and Tomatillo-Red Chili Salsa', 'Chips and Mild Fresh Tomato Salsa']
11
{'Chips and Fresh Tomato Salsa': [130.0], 'Chips and Tomatillo-Green Chili Salsa': [33.0], 'Side of Chips': [110.0], 'Chips and Guacamole': [506.0], 'Chips and Tomatillo Green Chili Salsa': [45.0], 'Chips': [230.0], 'Chips and Tomatillo Red Chili Salsa': [50.0], 'Chips and Roasted Chili-Corn Salsa': [18.0], 'Chips and Roasted Chili Corn Salsa': [23.0], 'Chips and Tomatillo-Red Chili Salsa': [25.0], 'Chips and Mild Fresh Tomato Salsa': [1.0]}


In [24]:
# sorting chips count all from lowest value to highest value
sorted_chips_all = sorted(chips_count_all.items(), key=operator.itemgetter(1))
print (sorted_chips_all)

[('Chips and Mild Fresh Tomato Salsa', [1.0]), ('Chips and Roasted Chili-Corn Salsa', [18.0]), ('Chips and Roasted Chili Corn Salsa', [23.0]), ('Chips and Tomatillo-Red Chili Salsa', [25.0]), ('Chips and Tomatillo-Green Chili Salsa', [33.0]), ('Chips and Tomatillo Green Chili Salsa', [45.0]), ('Chips and Tomatillo Red Chili Salsa', [50.0]), ('Side of Chips', [110.0]), ('Chips and Fresh Tomato Salsa', [130.0]), ('Chips', [230.0]), ('Chips and Guacamole', [506.0])]


The most popular order for chips -- by a very long shot -- is **Chips with Guacamole**: **506** vs. **just Chips** in second place with **230** orders

In [25]:
# now using .defaultdict() because I did not use that in the example above

from collections import defaultdict

chips_list = [] #store the entire list of chips here, i.e. no need to filter out duplicates because we'll count on that basis
for row in split_data:
    if "Chips" in row[2]:
        chips_list.append(row[2])

chips_count = defaultdict(int) # default value is zero 0
for chips in chips_list:
    chips_count[chips] += 1 # increments by values 1 every time it finds the same order

chips_count # call up the list by name spits out list of Chips orders sorted by key [equivalent to itemgetter(0)]

defaultdict(int,
            {'Chips': 211,
             'Chips and Fresh Tomato Salsa': 110,
             'Chips and Guacamole': 479,
             'Chips and Mild Fresh Tomato Salsa': 1,
             'Chips and Roasted Chili Corn Salsa': 22,
             'Chips and Roasted Chili-Corn Salsa': 18,
             'Chips and Tomatillo Green Chili Salsa': 43,
             'Chips and Tomatillo Red Chili Salsa': 48,
             'Chips and Tomatillo-Green Chili Salsa': 31,
             'Chips and Tomatillo-Red Chili Salsa': 20,
             'Side of Chips': 101})

In [26]:
# to make it easier to see smallest vs. largest item asked for, it's better to sort. Note: if itemgetter(0), it sorts by key
sorted_chips = sorted(chips_count.items(),key=operator.itemgetter(1))
sorted_chips

[('Chips and Mild Fresh Tomato Salsa', 1),
 ('Chips and Roasted Chili-Corn Salsa', 18),
 ('Chips and Tomatillo-Red Chili Salsa', 20),
 ('Chips and Roasted Chili Corn Salsa', 22),
 ('Chips and Tomatillo-Green Chili Salsa', 31),
 ('Chips and Tomatillo Green Chili Salsa', 43),
 ('Chips and Tomatillo Red Chili Salsa', 48),
 ('Side of Chips', 101),
 ('Chips and Fresh Tomato Salsa', 110),
 ('Chips', 211),
 ('Chips and Guacamole', 479)]

In [ ]:
# saw a Counter() function but for some reason cannot use it here...why?
# https://docs.python.org/3/library/collections.html

---

## Bonus: Craft a problem statement about this data that interests you, and then answer it!


### Message for you:
**I already did a side project before I read through the 6th section and the bottom of the page here. **
** Please have a look at the function that counts orders of different soda / canned soda types above.**

In [ ]:
# If I can think of another cool analysis, I'll include that here.